common/lib/xmodule/xmodule/contentstore/content.py

```diff
@@ -351,27 +367,30 @@ def generate_thumbnail(self, content, tempfile_path=None, dimensions=None):
                 else:
                     im = Image.open(tempfile_path)
 
-                # I've seen some exceptions from the PIL library when trying to save palletted
-                # PNG files to JPEG. Per the google-universe, they suggest converting to RGB first.
-                im = im.convert('RGB')
+                # We use the context processor here to avoid leaking the inner file descriptor
+                # of the Image object -- this way it gets closed after we're done with using it.
+                thumbnail_file = StringIO.StringIO()
+                with im:
```

Comment:
Yes, see https://docs.python.org/2.5/whatsnew/pep-343.html:

> Let's assume there's an object representing a database connection. Our goal will be to let the user write code like this:
```
db_connection = DatabaseConnection()
with db_connection as cursor:
```

Also my test case verifies that `__exit__` and thus `close()` is called.

I did it this way because what we pass to `Image.open` depends on whether `tempfile_path` is None, so it was easiest to change this way. However it could be indeed written so that the constructor's argument is extracted in a separate variable, to avoid reusing the `im` name outside the `with` statement. I'll change the code as preferred by @nedbat.